lesson08.md

```diff
@@ -0,0 +1,364 @@
+
+So, what's a function?
+
+A function is a reusable block of code.
+<!-- .element: class="fragment" -->
+
+The same way in which a variable contains a value, a function contains `code`.
+<!-- .element: class="fragment" -->
+
+---
+
+### Functions: execution
+
+Code is normally executed by JavaScript.
+
+When some code is inside a function, we have to explicity "call" that function in order to execute the code inside.
+
+How do we call a function?
+
+We write the name of the function followed by open and closed parenthesis.
+<!-- .element: class="fragment" -->
+
+---
+
+### Exercise 
+
+1. create a function called `sum`
+1. this function can take in input two parameters called `a` and `b`
+1. inside the function, calculate the sum of `a` and `b` and save it in a variable called `result`
+1. return `result`
+1. call the function with two numerical values and save the result in a variable
+1. print the result to the console
+
+BONUS: if you finish in time, do the same with a function `multiply`
+
+---
+
+### Function: recap quiz 
+
+- how do we define a function?
+<!-- .element: class="fragment" -->
+- how do we call a function?
+<!-- .element: class="fragment" -->
+- what are parameters?
+<!-- .element: class="fragment" -->
+- how many parameters can a function have?
+<!-- .element: class="fragment" -->
+- how do we pass a value to parameters?
+<!-- .element: class="fragment" -->
+- how do we get a result from functions?
+<!-- .element: class="fragment" -->
+
+---
+
+### return statement 
+
+What does this print?
+
+```js
+function sum(a, b) {
+    let result = a + b;
+}
+
+let total = sum(4, 5);
+console.log("The total is:", total);
+```
+
+A function with no return statement, returns `undefined` by default.
+<!-- .element: class="fragment" -->
+
+---
+
+### return statement 
+
+What happens inside a function, stays inside the function.
+
+Unless we use return.
+<!-- .element: class="fragment" -->
+
+```js
+function sum(a, b) {
+    let result = a + b;
+    return result;
+}
+
+let total = sum(4, 5);
+console.log("The total is:", total);
+```
+
+---
+
+### return statement vs console.log()
+
+What's the value of `total` here?
+
+```js
+function sum(a, b) {
+    let result = a + b;
+    console.log(result);
+}
+
+let total = sum(4, 5);
+console.log("The total is:", total);
+```
+
+Total is `undefined`: `console.log()` is not the same as return, it's a way to check a value in the console.
+<!-- .element: class="fragment" -->
+
+---
+
+### return statement vs console.log()
+
+Very often we want to use the result of a function inside of another function.
+
+Let's combine the functions I've created so far:
+
+```js
+let total1 = sum(235, 485);
+let total2 = sum(27, 382);
+let total3 = sum(157, 211);
+let max = maximumOf3(total1, total2, total3); 
+
+console.log(max);
+```
+
+---
+
+### return statement vs console.log()
+
+If I didn't return a value from inside the `sum()` function, I wouldn't have been able to use it outside of it.
```

Comment:
```suggestion
Without a `return` in the `sum()` function, the value would be unavailable outside the function.
```

Reply:
Still some a lot of negatives :/ Maybe: "Only with a return inside sum(), the value is available outside the function" ??

Reply:
mh, you're right, let me think a bit about it